src/escpos/printer.py

```diff
@@ -84,6 +84,10 @@ def _raw(self, msg):
         """
         self.device.write(self.out_ep, msg, self.timeout)
 
+    def _read(self):
```

Comment:
You should add a dummy implementation to the base class that throws a NotImplementedError.